### PR TITLE
[CI][RL]: add end-to-end Verl+vLLM FSDP PPO test script for GSM8K fine-tuning

### PR DESCRIPTION
**Pull request by erfgss**
## Purpose

This PR adds an end-to-end example script for running **GSM8K PPO training with Verl + vLLM**, enabling a reproducible workflow that includes:

- Environment setup for Verl  
- GSM8K dataset preprocessing  
- PPO training using Qwen2.5-0.5B-Instruct  
- FSDP checkpoint merging into HuggingFace format  
- Evaluation using `lm_eval`  
- Fully automated steps in a single script  

This helps validate the PPO pipeline correctness and provides a ready-to-use example for developers and CI verification.

### 1. Run the full pipeline
```bash
bash run_gsm8k_ppo.sh 2>&1 | tee run_gsm8k_ppo.log
```

### 2. Script behavior
- Clone Verl (if not already present)
- Install Verl + vLLM extras via `uv`
- Preprocess GSM8K:
  ```bash
  python3 gsm8k.py --local_save_dir <data_dir>
  ```
- Download base model:
  ```bash
  huggingface-cli download Qwen/Qwen2.5-0.5B-Instruct      --resume-download --local-dir <MODEL_DIR>
  ```
- PPO training:
```bash
python3 -m verl.trainer.main_ppo   data.train_files="${data_dir}/train.parquet"   data.val_files="${data_dir}/train.parquet"   data.train_batch_size=256   data.max_prompt_length=512   data.max_response_length=512   actor_rollout_ref.model.path="${MODEL_DIR}"   actor_rollout_ref.actor.optim.lr=1e-6   actor_rollout_ref.actor.ppo_mini_batch_size=64   actor_rollout_ref.actor.ppo_micro_batch_size_per_gpu=4   actor_rollout_ref.rollout.name=vllm   actor_rollout_ref.rollout.log_prob_micro_batch_size_per_gpu=8   actor_rollout_ref.rollout.tensor_model_parallel_size=1   actor_rollout_ref.rollout.gpu_memory_utilization=0.4   actor_rollout_ref.ref.log_prob_micro_batch_size_per_gpu=4   critic.optim.lr=1e-5   critic.model.path="${MODEL_DIR}"   critic.ppo_micro_batch_size_per_gpu=4   algorithm.kl_ctrl.kl_coef=0.001   trainer.logger=tensorboard   trainer.val_before_train=False   trainer.n_gpus_per_node="${n_gpus_per_node}"   trainer.nnodes="${nnodes}"   trainer.save_freq=10   trainer.test_freq=10   trainer.total_epochs="${train_epochs}"
```
- Merge FSDP weights:
  ```bash
  python scripts/legacy_model_merger.py merge     --backend fsdp     --local_dir <LOCAL_DIR>     --target_dir <MERGED_DIR>
  ```
- Evaluate with lm-eval:
  ```bash
  CUDA_VISIBLE_DEVICES=0,1,2,3 lm_eval --model hf       --model_args pretrained=<MERGED_DIR>,trust_remote_code=True       --tasks gsm8k       --apply_chat_template True       --output_path results_ppo.json
  ```

## Test Result
Results saved to:
```
results_ppo.json
```
hf(pretrained=~/vllm/verl/examples/data_preprocess/checkpoints/verl_examples/gsm8k/global_step_29/actor_hf,trust_remote_code=True), gen_kwahot: None, batch_size: auto
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.3510|±  |0.0131|
|     |       |strict-match    |     5|exact_match|↑  |0.0136|±  |0.0032|

## Reference link:：
https://verl.readthedocs.io/en/latest/start/install.html
https://verl.readthedocs.io/en/latest/start/quickstart.html
---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

File: .buildkite/scripts/run_ppo_test.sh [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+python3 gsm8k.py --local_save_dir "${data_dir}"; 
+
+echo "===== gsm8k.py preprocessing completed! ====="
+
+echo "===== Downloading model: ${MODEL_ID} ====="
+echo "Target directory: ${MODEL_DIR}"
+huggingface-cli download "${MODEL_ID}" --resume-download --local-dir "${MODEL_DIR}"
+echo "===== Downloading model: ${MODEL_ID} completed! ====="
+echo "===== Starting PPO Training ====="
+python3 -m verl.trainer.main_ppo \
```
Comment (david6666666) [inline .buildkite/scripts/run_ppo_test.sh:63]: Thank you! Can you use Verl's scripts to prevent the test from failing if Verl modifies the parameters?

File: .buildkite/scripts/run_ppo_test.sh [hunk 2]
```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
```
Comment (david6666666) [inline .buildkite/scripts/run_ppo_test.sh:1]: Rename the file to run-verl-ppo-test.sh

**Comment by erfgss** [comment]
@youkaichao @22quinn PTAL, Thanks

File: .buildkite/scripts/run_ppo_test.sh [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+if ! command -v uv &> /dev/null; then
+    echo "Installing UV package manager..."
+    curl -LsSf https://astral.sh/uv/install.sh | sh
+    source $HOME/.local/bin/env
+fi
+
+echo "Entering ${VERL_DIR} ..."
+cd "${VERL_DIR}"
+uv pip install --no-deps -e .
+uv pip install -e .[vllm]
```
Comment (mgoin) [inline .buildkite/scripts/run_ppo_test.sh:49]: Currently this script seems like it isn't meant to run in CI, since you are overriding the vllm version here (I assume). Could this be setup to assume you have already installed vllm to test main or a branch?

File: .buildkite/scripts/run_ppo_test.sh [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+VERL_REPO="https://github.com/volcengine/verl.git"
+VERL_BRANCH="main"
+VERL_DIR="${REPO_ROOT}/verl"
+TARGET_DIR="${VERL_DIR}/examples/data_preprocess"
+MODEL_ID=Qwen/Qwen2.5-0.5B-Instruct
+MODEL_DIR="${VERL_DIR}/models/Qwen2.5-0.5B-Instruct"
+train_epochs=2
+data_dir="${VERL_DIR}/gsm8k"
+n_gpus_per_node=8
```
Comment (mgoin) [inline .buildkite/scripts/run_ppo_test.sh:17]: Does it require 8 GPUs? We have some 4 GPU H100 runners already that we could run this on nightly
